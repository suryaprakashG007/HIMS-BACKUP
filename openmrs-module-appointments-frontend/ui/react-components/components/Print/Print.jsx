import React, { useState } from 'react'
import { createPortal } from 'react-dom'

export default function Print({ children, ...props }) {
    const [printRef, setPrintRef] = useState(null)
    const mountNode = printRef && printRef.contentWindow.document.body

    function handlePrint() {
        printRef.contentWindow.print();

        return false;
    }

    return (
        <>
            <iframe {...props} ref={setPrintRef} style={{ display: 'none' }} data-testid="printFrame">
                {mountNode &&
                    createPortal(
                        React.Children.only(children),
                        mountNode
                    )}
            </iframe>
            <a onClick={() => handlePrint()} data-testid="printIcon"><i className="fas fa-print"></i></a>
        </>
    )
}