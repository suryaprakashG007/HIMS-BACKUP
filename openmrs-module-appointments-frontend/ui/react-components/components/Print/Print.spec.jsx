import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Print from './Print'

describe('Print', () => {
    it('render print icon', () => {
        const { asFragment } = render(<Print><div>Sample content</div></Print>)
        expect(asFragment()).toMatchSnapshot()
    })

    it('print method is invoked on iframe on click of icon', () => {
        const {getByTestId} = render(<Print><div>Sample content</div></Print>)
        getByTestId("printFrame").contentWindow.print = jest.fn()
        fireEvent.click(getByTestId("printIcon"))

        expect(getByTestId("printFrame").contentWindow.print).toHaveBeenCalled()
    })
})