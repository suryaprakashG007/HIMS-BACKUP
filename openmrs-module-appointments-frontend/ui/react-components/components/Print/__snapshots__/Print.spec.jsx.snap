// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print render print icon 1`] = `
<DocumentFragment>
  <iframe
    data-testid="printFrame"
    style="display: none;"
  />
  <a
    data-testid="printIcon"
  >
    <i
      class="fas fa-print"
    />
  </a>
</DocumentFragment>
`;
